File: src/features/typing/data/java/jump_game_ii.java
class Solution {
    public int jump(int[] nums) {
        int steps = 0;
        int end = 0;
        int farthest = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if (i + nums[i] > farthest) {
                farthest = i + nums[i];
            }
            if (i == end) {
                steps++;
                end = farthest;
            }
        }
        return steps;
    }
}

File: src/features/typing/data/ts/jump_game_ii.ts
function jump(nums: number[]): number {
    let steps = 0;
    let end = 0;
    let farthest = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        if (i + nums[i] > farthest) {
            farthest = i + nums[i];
        }
        if (i === end) {
            steps++;
            end = farthest;
        }
    }
    return steps;
}
